From 9f96e2b3ade2bf3ae188f9df4fc760f0f02479d4 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <tim.stahlhut@gmail.com>
Date: Fri, 4 Sep 2015 15:23:28 -0400
Subject: [PATCH] * cbp: Added lib folders to some CB projects. (Thanks
 stahta01)

Made sure that all the .a files are under lib folder using contrib sub-folder and sub-folders.
Removed some unneeded search folders.
Removed some unneeded libs.
---
 src/CodeBlocks.cbp                                 | 60 +++++++++----------
 src/CodeBlocks_wx30.cbp                            | 56 +++++++++---------
 src/CodeBlocks_wx30_64.cbp                         | 68 +++++++++++-----------
 src/plugins/contrib/SpellChecker/SpellChecker.cbp  |  5 +-
 .../contrib/SpellChecker/SpellChecker_wx30.cbp     |  5 +-
 .../contrib/SpellChecker/SpellChecker_wx30_64.cbp  |  5 +-
 src/plugins/contrib/ToolsPlus/ToolsPlus.cbp        |  1 -
 src/plugins/contrib/cb_koders/cb_koders.cbp        |  2 +-
 src/plugins/contrib/cb_koders/cb_koders_wx30.cbp   |  3 +-
 .../contrib/cb_koders/cb_koders_wx30_64.cbp        |  2 +-
 src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp |  4 +-
 .../contrib/devpak_plugin/DevPakPlugin_wx30.cbp    |  4 +-
 .../contrib/devpak_plugin/DevPakPlugin_wx30_64.cbp |  6 +-
 src/plugins/contrib/help_plugin/help-plugin.cbp    |  9 +--
 .../contrib/help_plugin/help-plugin_wx30.cbp       |  9 +--
 .../contrib/help_plugin/help-plugin_wx30_64.cbp    |  9 +--
 src/plugins/contrib/regex_testbed/RegExTestbed.cbp |  2 +-
 .../contrib/regex_testbed/RegExTestbed_wx30.cbp    |  2 +-
 .../contrib/regex_testbed/RegExTestbed_wx30_64.cbp |  2 +-
 src/plugins/contrib/source_exporter/Exporter.cbp   |  6 +-
 .../contrib/source_exporter/Exporter_wx30.cbp      |  6 +-
 .../contrib/source_exporter/Exporter_wx30_64.cbp   |  6 +-
 .../contrib/wxContribItems/wxContribItems.cbp      | 17 +++---
 .../contrib/wxContribItems/wxContribItems_wx30.cbp | 16 +++--
 .../wxContribItems/wxContribItems_wx30_64.cbp      | 14 ++---
 src/plugins/contrib/wxSmith/wxSmith.cbp            |  2 +-
 src/plugins/contrib/wxSmith/wxSmith_wx30.cbp       |  2 +-
 src/plugins/contrib/wxSmith/wxSmith_wx30_64.cbp    |  2 +-
 src/plugins/contrib/wxSmithPlot/wxsSmithPlot.cbp   |  3 +-
 src/tools/cb_share_config/cb_share_config.cbp      |  2 +-
 src/tools/cb_share_config/cb_share_config_wx30.cbp |  2 +-
 .../cb_share_config/cb_share_config_wx30_64.cbp    |  4 +-
 32 files changed, 169 insertions(+), 167 deletions(-)

diff --git a/src/CodeBlocks.cbp b/src/CodeBlocks.cbp
index 6b8b55a..dfa3bc6 100644
--- a/src/CodeBlocks.cbp
+++ b/src/CodeBlocks.cbp
@@ -7,7 +7,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="tinyXML">
-				<Option output="base/tinyxml/txml" prefix_auto="1" extension_auto="1" />
+				<Option output="lib/txml" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option type="2" />
 				<Option compiler="gcc" />
@@ -26,7 +26,7 @@
 				<Linker>
 					<Add option="-s" />
 					<Add library="txml" />
-					<Add directory="base/tinyxml" />
+					<Add directory="lib" />
 				</Linker>
 			</Target>
 			<Target title="ConsoleRunner">
@@ -40,7 +40,7 @@
 				</Compiler>
 			</Target>
 			<Target title="Squirrel">
-				<Option output="sdk/scripting/lib/squirrel" prefix_auto="1" extension_auto="1" />
+				<Option output="lib/squirrel" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option type="2" />
 				<Option compiler="gcc" />
@@ -51,9 +51,9 @@
 				</Compiler>
 			</Target>
 			<Target title="Squirrel std lib">
-				<Option output="sdk/scripting/lib/sqstdlib" prefix_auto="1" extension_auto="1" />
+				<Option output="lib/sqstdlib" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
-				<Option external_deps="sdk/scripting/lib/libsquirrel.a;" />
+				<Option external_deps="lib/libsquirrel.a;" />
 				<Option type="2" />
 				<Option compiler="gcc" />
 				<Compiler>
@@ -63,9 +63,9 @@
 				</Compiler>
 			</Target>
 			<Target title="SqPlus">
-				<Option output="sdk/scripting/lib/sqplus" prefix_auto="1" extension_auto="1" />
+				<Option output="lib/sqplus" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
-				<Option external_deps="sdk/scripting/lib/libsquirrel.a;sdk/scripting/lib/libsqstdlib.a;" />
+				<Option external_deps="lib/libsquirrel.a;lib/libsqstdlib.a;" />
 				<Option type="2" />
 				<Option compiler="gcc" />
 				<Compiler>
@@ -75,7 +75,7 @@
 				</Compiler>
 			</Target>
 			<Target title="scintilla">
-				<Option output="devel/wxscintilla_cb" prefix_auto="1" extension_auto="1" />
+				<Option output="lib/wxscintilla_cb" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option type="2" />
 				<Option compiler="gcc" />
@@ -101,7 +101,7 @@
 				</Linker>
 			</Target>
 			<Target title="wxpropgrid">
-				<Option output="devel/wxpropgrid" prefix_auto="1" extension_auto="1" />
+				<Option output="devel/wxpropgrid" imp_lib="lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -123,9 +123,9 @@
 				</Linker>
 			</Target>
 			<Target title="sdk">
-				<Option output="devel/codeblocks" prefix_auto="1" extension_auto="1" />
+				<Option output="devel/codeblocks" imp_lib="lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="base/tinyxml/libtxml.a;sdk/scripting/lib/libsquirrel.a;sdk/scripting/lib/libsqstdlib.a;sdk/scripting/lib/libsqplus.a;devel/libwxscintilla_cb.a;" />
+				<Option external_deps="lib/libtxml.a;lib/libsquirrel.a;lib/libsqstdlib.a;lib/libsqplus.a;lib/libwxscintilla_cb.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --verbose -p debug" />
@@ -163,9 +163,8 @@
 					<Add library="sqstdlib" />
 					<Add library="squirrel" />
 					<Add library="gdi32" />
+					<Add directory="lib" />
 					<Add directory="devel" />
-					<Add directory="sdk/scripting/lib" />
-					<Add directory="sdk/propgrid" />
 				</Linker>
 				<ExtraCommands>
 					<Add before="build_tools\autorevision\autorevision +wx +int +t . include/autorevision.h" />
@@ -178,7 +177,7 @@
 			<Target title="src">
 				<Option output="devel/codeblocks" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="0" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -234,7 +233,7 @@
 			<Target title="Abbreviations">
 				<Option output="devel/share/CodeBlocks/plugins/abbreviations" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -267,7 +266,7 @@
 			<Target title="AStyle">
 				<Option output="devel/share/CodeBlocks/plugins/astyle" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -299,7 +298,7 @@
 			<Target title="Autosave">
 				<Option output="devel/share/CodeBlocks/plugins/autosave" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -327,7 +326,7 @@
 				</ExtraCommands>
 			</Target>
 			<Target title="Compiler depslib">
-				<Option output="plugins/compilergcc/depslib/depslib" prefix_auto="1" extension_auto="1" />
+				<Option output="lib/depslib" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option type="2" />
 				<Option compiler="gcc" />
@@ -342,7 +341,7 @@
 			<Target title="Compiler">
 				<Option output="devel/share/CodeBlocks/plugins/compiler" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;plugins/compilergcc/depslib/libdepslib.a;" />
+				<Option external_deps="lib/libcodeblocks.a;lib/libdepslib.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -366,8 +365,8 @@
 					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
 					<Add library="depslib" />
 					<Add library="wxpropgrid" />
+					<Add directory="lib" />
 					<Add directory="devel" />
-					<Add directory="plugins/compilergcc/depslib" />
 				</Linker>
 				<ExtraCommands>
 					<Add after="cmd /c if not exist devel\share\CodeBlocks mkdir devel\share\CodeBlocks" />
@@ -381,7 +380,7 @@
 			<Target title="Debugger">
 				<Option output="devel/share/CodeBlocks/plugins/debugger" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -413,7 +412,7 @@
 			<Target title="Code-completion">
 				<Option output="devel/share/CodeBlocks/plugins/codecompletion" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -449,7 +448,7 @@
 			<Target title="Class wizard">
 				<Option output="devel/share/CodeBlocks/plugins/classwizard" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -479,7 +478,7 @@
 			<Target title="Default MIME handler">
 				<Option output="devel/share/CodeBlocks/plugins/defaultmimehandler" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -509,7 +508,7 @@
 			<Target title="Occurrences Highlighting">
 				<Option output="devel/share/CodeBlocks/plugins/OccurrencesHighlighting" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -539,7 +538,7 @@
 			<Target title="Open files list">
 				<Option output="devel/share/CodeBlocks/plugins/openfileslist" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -569,7 +568,7 @@
 			<Target title="Projects-workspaces importer">
 				<Option output="devel/share/CodeBlocks/plugins/projectsimporter" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -605,7 +604,7 @@
 			<Target title="Scripted wizard">
 				<Option output="devel/share/CodeBlocks/plugins/scriptedwizard" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -637,7 +636,7 @@
 			<Target title="To-do">
 				<Option output="devel/share/CodeBlocks/plugins/todo" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -673,7 +672,7 @@
 			<Target title="XP look &amp; feel">
 				<Option output="devel/share/CodeBlocks/plugins/xpmanifest" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel" />
-				<Option external_deps="devel/libcodeblocks.a;" />
+				<Option external_deps="lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -738,7 +737,6 @@
 		</ResourceCompiler>
 		<Linker>
 			<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
-			<Add directory="base/tinyxml" />
 			<Add directory="$(#WX.lib)/gcc_dll$(WX_CFG)" />
 		</Linker>
 		<Unit filename="base/tinyxml/tinystr.cpp">
diff --git a/src/CodeBlocks_wx30.cbp b/src/CodeBlocks_wx30.cbp
index 3f63122..0857d1b 100644
--- a/src/CodeBlocks_wx30.cbp
+++ b/src/CodeBlocks_wx30.cbp
@@ -7,7 +7,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="tinyXML">
-				<Option output="base/tinyxml/txml" prefix_auto="1" extension_auto="1" />
+				<Option output="lib30/txml" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output=".objs30" />
 				<Option type="2" />
@@ -28,7 +28,7 @@
 				<Linker>
 					<Add option="-s" />
 					<Add library="txml" />
-					<Add directory="base/tinyxml" />
+					<Add directory="lib30" />
 				</Linker>
 			</Target>
 			<Target title="ConsoleRunner">
@@ -43,7 +43,7 @@
 				</Compiler>
 			</Target>
 			<Target title="Squirrel">
-				<Option output="sdk/scripting/lib/squirrel" prefix_auto="1" extension_auto="1" />
+				<Option output="lib30/squirrel" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output=".objs30" />
 				<Option type="2" />
@@ -55,10 +55,10 @@
 				</Compiler>
 			</Target>
 			<Target title="Squirrel std lib">
-				<Option output="sdk/scripting/lib/sqstdlib" prefix_auto="1" extension_auto="1" />
+				<Option output="lib30/sqstdlib" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output=".objs30" />
-				<Option external_deps="sdk/scripting/lib/libsquirrel.a;" />
+				<Option external_deps="lib30/libsquirrel.a;" />
 				<Option type="2" />
 				<Option compiler="gcc" />
 				<Compiler>
@@ -68,10 +68,10 @@
 				</Compiler>
 			</Target>
 			<Target title="SqPlus">
-				<Option output="sdk/scripting/lib/sqplus" prefix_auto="1" extension_auto="1" />
+				<Option output="lib30/sqplus" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output=".objs30" />
-				<Option external_deps="sdk/scripting/lib/libsquirrel.a;sdk/scripting/lib/libsqstdlib.a;" />
+				<Option external_deps="lib30/libsquirrel.a;lib30/libsqstdlib.a;" />
 				<Option type="2" />
 				<Option compiler="gcc" />
 				<Compiler>
@@ -81,7 +81,7 @@
 				</Compiler>
 			</Target>
 			<Target title="scintilla">
-				<Option output="devel30/wxscintilla_cb" prefix_auto="1" extension_auto="1" />
+				<Option output="lib30/wxscintilla_cb" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output=".objs30" />
 				<Option type="2" />
@@ -108,10 +108,10 @@
 				</Linker>
 			</Target>
 			<Target title="sdk">
-				<Option output="devel30/codeblocks" prefix_auto="1" extension_auto="1" />
+				<Option output="devel30/codeblocks" imp_lib="lib30/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="base/tinyxml/libtxml.a;sdk/scripting/lib/libsquirrel.a;sdk/scripting/lib/libsqstdlib.a;sdk/scripting/lib/libsqplus.a;devel30/libwxscintilla_cb.a;" />
+				<Option external_deps="lib30/libtxml.a;lib30/libsquirrel.a;lib30/libsqstdlib.a;lib30/libsqplus.a;lib30/libwxscintilla_cb.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --verbose -p debug" />
@@ -147,8 +147,8 @@
 					<Add library="sqstdlib" />
 					<Add library="squirrel" />
 					<Add library="gdi32" />
+					<Add directory="lib30" />
 					<Add directory="devel30" />
-					<Add directory="sdk/scripting/lib" />
 				</Linker>
 				<ExtraCommands>
 					<Add before="build_tools/autorevision/autorevision +wx +int +t . include/autorevision.h" />
@@ -162,7 +162,7 @@
 				<Option output="devel30/codeblocks" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="devel30/libcodeblocks.a;" />
+				<Option external_deps="lib30/libcodeblocks.a;" />
 				<Option type="0" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -218,7 +218,7 @@
 				<Option output="devel30/share/CodeBlocks/plugins/abbreviations" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="devel30/libcodeblocks.a;" />
+				<Option external_deps="lib30/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -252,7 +252,7 @@
 				<Option output="devel30/share/CodeBlocks/plugins/astyle" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="devel30/libcodeblocks.a;" />
+				<Option external_deps="lib30/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -285,7 +285,7 @@
 				<Option output="devel30/share/CodeBlocks/plugins/autosave" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="devel30/libcodeblocks.a;" />
+				<Option external_deps="lib30/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -313,7 +313,7 @@
 				</ExtraCommands>
 			</Target>
 			<Target title="Compiler depslib">
-				<Option output="plugins/compilergcc/depslib/depslib" prefix_auto="1" extension_auto="1" />
+				<Option output="lib30/depslib" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output=".objs30" />
 				<Option type="2" />
@@ -330,7 +330,7 @@
 				<Option output="devel30/share/CodeBlocks/plugins/compiler" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="devel30/libcodeblocks.a;plugins/compilergcc/depslib/libdepslib.a;" />
+				<Option external_deps="lib30/libcodeblocks.a;lib30/libdepslib.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -353,8 +353,8 @@
 					<Add library="codeblocks" />
 					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
 					<Add library="depslib" />
+					<Add directory="lib30" />
 					<Add directory="devel30" />
-					<Add directory="plugins/compilergcc/depslib" />
 				</Linker>
 				<ExtraCommands>
 					<Add after="cmd /c if not exist devel30\share\CodeBlocks mkdir devel30\share\CodeBlocks" />
@@ -369,7 +369,7 @@
 				<Option output="devel30/share/CodeBlocks/plugins/debugger" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="devel30/libcodeblocks.a;" />
+				<Option external_deps="lib30/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -402,7 +402,7 @@
 				<Option output="devel30/share/CodeBlocks/plugins/codecompletion" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="devel30/libcodeblocks.a;" />
+				<Option external_deps="lib30/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -440,7 +440,7 @@
 				<Option output="devel30/share/CodeBlocks/plugins/classwizard" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="devel30/libcodeblocks.a;" />
+				<Option external_deps="lib30/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -471,7 +471,7 @@
 				<Option output="devel30/share/CodeBlocks/plugins/defaultmimehandler" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="devel30/libcodeblocks.a;" />
+				<Option external_deps="lib30/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -502,7 +502,7 @@
 				<Option output="devel30/share/CodeBlocks/plugins/OccurrencesHighlighting" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="devel30/libcodeblocks.a;" />
+				<Option external_deps="lib30/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -533,7 +533,7 @@
 				<Option output="devel30/share/CodeBlocks/plugins/openfileslist" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="devel30/libcodeblocks.a;" />
+				<Option external_deps="lib30/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -564,7 +564,7 @@
 				<Option output="devel30/share/CodeBlocks/plugins/projectsimporter" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="devel30/libcodeblocks.a;" />
+				<Option external_deps="lib30/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -601,7 +601,7 @@
 				<Option output="devel30/share/CodeBlocks/plugins/scriptedwizard" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="devel30/libcodeblocks.a;" />
+				<Option external_deps="lib30/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -634,7 +634,7 @@
 				<Option output="devel30/share/CodeBlocks/plugins/todo" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="devel30/libcodeblocks.a;" />
+				<Option external_deps="lib30/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -671,7 +671,7 @@
 				<Option output="devel30/share/CodeBlocks/plugins/xpmanifest" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30" />
 				<Option object_output=".objs30" />
-				<Option external_deps="devel30/libcodeblocks.a;" />
+				<Option external_deps="lib30/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
diff --git a/src/CodeBlocks_wx30_64.cbp b/src/CodeBlocks_wx30_64.cbp
index afea144..481b8a4 100644
--- a/src/CodeBlocks_wx30_64.cbp
+++ b/src/CodeBlocks_wx30_64.cbp
@@ -7,7 +7,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="tinyXML">
-				<Option output="base/tinyxml/txml30_64" prefix_auto="1" extension_auto="1" />
+				<Option output="lib30_64/txml" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output=".objs30_64" />
 				<Option type="2" />
@@ -27,8 +27,8 @@
 				</Compiler>
 				<Linker>
 					<Add option="-s" />
-					<Add library="txml30_64" />
-					<Add directory="base/tinyxml" />
+					<Add library="txml" />
+					<Add directory="lib30_64" />
 				</Linker>
 			</Target>
 			<Target title="ConsoleRunner">
@@ -43,7 +43,7 @@
 				</Compiler>
 			</Target>
 			<Target title="Squirrel">
-				<Option output="sdk/scripting/lib/squirrel30_64" prefix_auto="1" extension_auto="1" />
+				<Option output="lib30_64/squirrel" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output=".objs30_64" />
 				<Option type="2" />
@@ -55,10 +55,10 @@
 				</Compiler>
 			</Target>
 			<Target title="Squirrel std lib">
-				<Option output="sdk/scripting/lib/sqstdlib30_64" prefix_auto="1" extension_auto="1" />
+				<Option output="lib30_64/sqstdlib" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="sdk/scripting/lib/libsquirrel30_64.a;" />
+				<Option external_deps="lib30_64/libsquirrel.a;" />
 				<Option type="2" />
 				<Option compiler="gcc" />
 				<Compiler>
@@ -68,10 +68,10 @@
 				</Compiler>
 			</Target>
 			<Target title="SqPlus">
-				<Option output="sdk/scripting/lib/sqplus30_64" prefix_auto="1" extension_auto="1" />
+				<Option output="lib30_64/sqplus" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="sdk/scripting/lib/libsquirrel30_64.a;sdk/scripting/lib/libsqstdlib30_64.a;" />
+				<Option external_deps="lib30_64/libsquirrel.a;lib30_64/libsqstdlib.a;" />
 				<Option type="2" />
 				<Option compiler="gcc" />
 				<Compiler>
@@ -81,7 +81,7 @@
 				</Compiler>
 			</Target>
 			<Target title="scintilla">
-				<Option output="devel30_64/wxscintilla_cb" prefix_auto="1" extension_auto="1" />
+				<Option output="lib30_64/wxscintilla_cb" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output=".objs30_64" />
 				<Option type="2" />
@@ -108,10 +108,10 @@
 				</Linker>
 			</Target>
 			<Target title="sdk">
-				<Option output="devel30_64/codeblocks" prefix_auto="1" extension_auto="1" />
+				<Option output="devel30_64/codeblocks" imp_lib="lib30_64/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="base/tinyxml/libtxml30_64.a;sdk/scripting/lib/libsquirrel30_64.a;sdk/scripting/lib/libsqstdlib30_64.a;sdk/scripting/lib/libsqplus30_64.a;devel30_64/libwxscintilla_cb.a;" />
+				<Option external_deps="lib30_64/libtxml.a;lib30_64/libsquirrel.a;lib30_64/libsqstdlib.a;lib30_64/libsqplus.a;lib30_64/libwxscintilla_cb.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --verbose -p debug" />
@@ -141,14 +141,14 @@
 					<Add option="-Wl,--enable-auto-import" />
 					<Add option="-Wl,--no-undefined" />
 					<Add library="shfolder" />
-					<Add library="txml30_64" />
+					<Add library="txml" />
 					<Add library="wxscintilla_cb" />
-					<Add library="sqplus30_64" />
-					<Add library="sqstdlib30_64" />
-					<Add library="squirrel30_64" />
+					<Add library="sqplus" />
+					<Add library="sqstdlib" />
+					<Add library="squirrel" />
 					<Add library="gdi32" />
+					<Add directory="lib30_64" />
 					<Add directory="devel30_64" />
-					<Add directory="sdk/scripting/lib" />
 				</Linker>
 				<ExtraCommands>
 					<Add before="build_tools/autorevision/autorevision30_64 +wx +int +t . include/autorevision.h" />
@@ -162,7 +162,7 @@
 				<Option output="devel30_64/codeblocks" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="devel30_64/libcodeblocks.a;" />
+				<Option external_deps="lib30_64/libcodeblocks.a;" />
 				<Option type="0" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -218,7 +218,7 @@
 				<Option output="devel30_64/share/CodeBlocks/plugins/abbreviations" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="devel30_64/libcodeblocks.a;" />
+				<Option external_deps="lib30_64/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -252,7 +252,7 @@
 				<Option output="devel30_64/share/CodeBlocks/plugins/astyle" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="devel30_64/libcodeblocks.a;" />
+				<Option external_deps="lib30_64/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -285,7 +285,7 @@
 				<Option output="devel30_64/share/CodeBlocks/plugins/autosave" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="devel30_64/libcodeblocks.a;" />
+				<Option external_deps="lib30_64/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -313,7 +313,7 @@
 				</ExtraCommands>
 			</Target>
 			<Target title="Compiler depslib">
-				<Option output="plugins/compilergcc/depslib/depslib30_64" prefix_auto="1" extension_auto="1" />
+				<Option output="lib30_64/depslib" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output=".objs30_64" />
 				<Option type="2" />
@@ -330,7 +330,7 @@
 				<Option output="devel30_64/share/CodeBlocks/plugins/compiler" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="devel30_64/libcodeblocks.a;plugins/compilergcc/depslib/libdepslib30_64.a;" />
+				<Option external_deps="lib30_64/libcodeblocks.a;lib30_64/libdepslib.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -352,9 +352,9 @@
 					<Add option="-mthreads" />
 					<Add library="codeblocks" />
 					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
-					<Add library="depslib30_64" />
+					<Add library="depslib" />
 					<Add directory="devel30_64" />
-					<Add directory="plugins/compilergcc/depslib" />
+					<Add directory="lib30_64" />
 				</Linker>
 				<ExtraCommands>
 					<Add after="cmd /c if not exist devel30_64\share\CodeBlocks mkdir devel30_64\share\CodeBlocks" />
@@ -369,7 +369,7 @@
 				<Option output="devel30_64/share/CodeBlocks/plugins/debugger" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="devel30_64/libcodeblocks.a;" />
+				<Option external_deps="lib30_64/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -402,7 +402,7 @@
 				<Option output="devel30_64/share/CodeBlocks/plugins/codecompletion" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="devel30_64/libcodeblocks.a;" />
+				<Option external_deps="lib30_64/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -440,7 +440,7 @@
 				<Option output="devel30_64/share/CodeBlocks/plugins/classwizard" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="devel30_64/libcodeblocks.a;" />
+				<Option external_deps="lib30_64/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -471,7 +471,7 @@
 				<Option output="devel30_64/share/CodeBlocks/plugins/defaultmimehandler" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="devel30_64/libcodeblocks.a;" />
+				<Option external_deps="lib30_64/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -502,7 +502,7 @@
 				<Option output="devel30_64/share/CodeBlocks/plugins/OccurrencesHighlighting" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="devel30_64/libcodeblocks.a;" />
+				<Option external_deps="lib30_64/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -533,7 +533,7 @@
 				<Option output="devel30_64/share/CodeBlocks/plugins/openfileslist" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="devel30_64/libcodeblocks.a;" />
+				<Option external_deps="lib30_64/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -564,7 +564,7 @@
 				<Option output="devel30_64/share/CodeBlocks/plugins/projectsimporter" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="devel30_64/libcodeblocks.a;" />
+				<Option external_deps="lib30_64/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -601,7 +601,7 @@
 				<Option output="devel30_64/share/CodeBlocks/plugins/scriptedwizard" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="devel30_64/libcodeblocks.a;" />
+				<Option external_deps="lib30_64/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -634,7 +634,7 @@
 				<Option output="devel30_64/share/CodeBlocks/plugins/todo" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="devel30_64/libcodeblocks.a;" />
+				<Option external_deps="lib30_64/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -671,7 +671,7 @@
 				<Option output="devel30_64/share/CodeBlocks/plugins/xpmanifest" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="devel30_64" />
 				<Option object_output=".objs30_64" />
-				<Option external_deps="devel30_64/libcodeblocks.a;" />
+				<Option external_deps="lib30_64/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
diff --git a/src/plugins/contrib/SpellChecker/SpellChecker.cbp b/src/plugins/contrib/SpellChecker/SpellChecker.cbp
index 241ae47..ae48302 100644
--- a/src/plugins/contrib/SpellChecker/SpellChecker.cbp
+++ b/src/plugins/contrib/SpellChecker/SpellChecker.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="hunspell">
-				<Option output="../../../devel/hunspell" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../lib/contrib/SpellChecker/hunspell" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs/plugins/contrib/SpellChecker/hunspell" />
 				<Option type="2" />
@@ -34,7 +34,7 @@
 				</Linker>
 			</Target>
 			<Target title="wxSpellChecker">
-				<Option output="../../../devel/wxSpellChecker" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../lib/contrib/SpellChecker/wxSpellChecker" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs/plugins/contrib/SpellChecker/wxSpellChecker" />
 				<Option type="2" />
@@ -91,6 +91,7 @@
 					<Add library="hunspell" />
 					<Add library="codeblocks" />
 					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
+					<Add directory="../../../lib/contrib/SpellChecker" />
 					<Add directory="../../../devel" />
 					<Add directory="$(#wx.lib)/gcc_dll$(WX_CFG)" />
 				</Linker>
diff --git a/src/plugins/contrib/SpellChecker/SpellChecker_wx30.cbp b/src/plugins/contrib/SpellChecker/SpellChecker_wx30.cbp
index 446189c..7c7d2c6 100644
--- a/src/plugins/contrib/SpellChecker/SpellChecker_wx30.cbp
+++ b/src/plugins/contrib/SpellChecker/SpellChecker_wx30.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="hunspell">
-				<Option output="../../../devel30/hunspell" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../lib30/contrib/SpellChecker/hunspell" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs30/plugins/contrib/SpellChecker/hunspell" />
 				<Option type="2" />
@@ -33,7 +33,7 @@
 				</Linker>
 			</Target>
 			<Target title="wxSpellChecker">
-				<Option output="../../../devel30/wxSpellChecker" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../lib30/contrib/SpellChecker/wxSpellChecker" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs30/plugins/contrib/SpellChecker/wxSpellChecker" />
 				<Option type="2" />
@@ -88,6 +88,7 @@
 					<Add library="hunspell" />
 					<Add library="codeblocks" />
 					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
+					<Add directory="../../../lib30/contrib/SpellChecker" />
 					<Add directory="../../../devel30" />
 					<Add directory="$(#WX30.lib)/gcc_dll$(WX_CFG)" />
 				</Linker>
diff --git a/src/plugins/contrib/SpellChecker/SpellChecker_wx30_64.cbp b/src/plugins/contrib/SpellChecker/SpellChecker_wx30_64.cbp
index 36597d4..2d62842 100644
--- a/src/plugins/contrib/SpellChecker/SpellChecker_wx30_64.cbp
+++ b/src/plugins/contrib/SpellChecker/SpellChecker_wx30_64.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="hunspell">
-				<Option output="../../../devel30_64/hunspell" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../lib30_64/contrib/SpellChecker/hunspell" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/SpellChecker/hunspell" />
 				<Option type="2" />
@@ -36,7 +36,7 @@
 				</Linker>
 			</Target>
 			<Target title="wxSpellChecker">
-				<Option output="../../../devel30_64/wxSpellChecker" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../lib30_64/contrib/SpellChecker/wxSpellChecker" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/SpellChecker/wxSpellChecker" />
 				<Option type="2" />
@@ -99,6 +99,7 @@
 					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
 					<Add directory="../../../devel30_64" />
 					<Add directory="$(#WX30_64.lib)/gcc_dll$(WX_CFG)" />
+					<Add directory="../../../lib30_64/contrib/SpellChecker" />
 				</Linker>
 				<ExtraCommands>
 					<Add after="cmd /c if not exist ..\..\..\devel30_64\share\CodeBlocks\SpellChecker mkdir ..\..\..\devel30_64\share\CodeBlocks\SpellChecker" />
diff --git a/src/plugins/contrib/ToolsPlus/ToolsPlus.cbp b/src/plugins/contrib/ToolsPlus/ToolsPlus.cbp
index 002d635..5692c34 100644
--- a/src/plugins/contrib/ToolsPlus/ToolsPlus.cbp
+++ b/src/plugins/contrib/ToolsPlus/ToolsPlus.cbp
@@ -18,7 +18,6 @@
 				</Compiler>
 				<Linker>
 					<Add library="codeblocks" />
-					<Add library="wxscintilla_cb" />
 					<Add library="wxmsw28$(WX_SUFFIX)" />
 					<Add library="gdi32" />
 					<Add directory="../../../devel" />
diff --git a/src/plugins/contrib/cb_koders/cb_koders.cbp b/src/plugins/contrib/cb_koders/cb_koders.cbp
index ef561b0..fdbc8b7 100644
--- a/src/plugins/contrib/cb_koders/cb_koders.cbp
+++ b/src/plugins/contrib/cb_koders/cb_koders.cbp
@@ -10,7 +10,7 @@
 			<Target title="default">
 				<Option output="../../../devel/share/CodeBlocks/plugins/cb_koders.dll" prefix_auto="0" extension_auto="0" />
 				<Option object_output="../../../.objs/plugins/contrib/cb_koders" />
-				<Option external_deps="../../../devel/libcodeblocks.a;../../../devel/libwxscintilla_cb.a;" />
+				<Option external_deps="../../../lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --multiple-instance -na -ns -nd -p debug" />
diff --git a/src/plugins/contrib/cb_koders/cb_koders_wx30.cbp b/src/plugins/contrib/cb_koders/cb_koders_wx30.cbp
index 7a32444..fca697f 100644
--- a/src/plugins/contrib/cb_koders/cb_koders_wx30.cbp
+++ b/src/plugins/contrib/cb_koders/cb_koders_wx30.cbp
@@ -9,7 +9,7 @@
 			<Target title="default">
 				<Option output="../../../devel30/share/CodeBlocks/plugins/cb_koders.dll" prefix_auto="0" extension_auto="0" />
 				<Option object_output="../../../.objs30/plugins/contrib/cb_koders" />
-				<Option external_deps="../../../devel30/libcodeblocks.a;" />
+				<Option external_deps="../../../lib30/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --multiple-instance -na -ns -nd -p debug" />
@@ -54,6 +54,7 @@
 			<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
 			<Add library="codeblocks" />
 			<Add directory="$(#WX30.lib)/gcc_dll$(WX_CFG)" />
+			<Add directory="../../../lib30" />
 			<Add directory="../../../devel30" />
 		</Linker>
 		<ExtraCommands>
diff --git a/src/plugins/contrib/cb_koders/cb_koders_wx30_64.cbp b/src/plugins/contrib/cb_koders/cb_koders_wx30_64.cbp
index 09baa8b..21f0b4a 100644
--- a/src/plugins/contrib/cb_koders/cb_koders_wx30_64.cbp
+++ b/src/plugins/contrib/cb_koders/cb_koders_wx30_64.cbp
@@ -9,7 +9,7 @@
 			<Target title="default">
 				<Option output="../../../devel30_64/share/CodeBlocks/plugins/cb_koders.dll" prefix_auto="0" extension_auto="0" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/cb_koders" />
-				<Option external_deps="../../../devel30_64/libcodeblocks.a;../../../devel30_64/libwxscintilla_cb.a;" />
+				<Option external_deps="../../../lib30_64/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --multiple-instance -na -ns -nd -p debug" />
diff --git a/src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp b/src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp
index a691278..433f320 100644
--- a/src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp
+++ b/src/plugins/contrib/devpak_plugin/DevPakPlugin.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="libbz2">
-				<Option output="bzip2/libbz2.a" prefix_auto="0" extension_auto="0" />
+				<Option output="../../../lib/contrib/devpak_plugin/libbz2.a" prefix_auto="0" extension_auto="0" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs/plugins/contrib/devpak_plugin" />
 				<Option type="2" />
@@ -48,6 +48,7 @@
 				<Linker>
 					<Add option="-mthreads" />
 					<Add library="bz2" />
+					<Add directory="../../../lib/contrib/devpak_plugin" />
 					<Add directory="../../../devel" />
 				</Linker>
 				<ExtraCommands>
@@ -75,7 +76,6 @@
 			<Add option="-Wl,--enable-auto-import" />
 			<Add library="codeblocks" />
 			<Add library="wxmsw28$(WX_SUFFIX)" />
-			<Add directory="bzip2" />
 			<Add directory="$(#WX.lib)/gcc_dll$(WX_CFG)" />
 		</Linker>
 		<Unit filename="bzip2/blocksort.c">
diff --git a/src/plugins/contrib/devpak_plugin/DevPakPlugin_wx30.cbp b/src/plugins/contrib/devpak_plugin/DevPakPlugin_wx30.cbp
index d1587c0..db7e802 100644
--- a/src/plugins/contrib/devpak_plugin/DevPakPlugin_wx30.cbp
+++ b/src/plugins/contrib/devpak_plugin/DevPakPlugin_wx30.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="libbz2">
-				<Option output="bzip2/libbz2.a" prefix_auto="0" extension_auto="0" />
+				<Option output="../../../lib30/contrib/devpak_plugin/libbz2.a" prefix_auto="0" extension_auto="0" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs30/plugins/contrib/devpak_plugin" />
 				<Option type="2" />
@@ -46,6 +46,7 @@
 				<Linker>
 					<Add option="-mthreads" />
 					<Add library="bz2" />
+					<Add directory="../../../lib30/contrib/devpak_plugin" />
 					<Add directory="../../../devel30" />
 				</Linker>
 				<ExtraCommands>
@@ -74,7 +75,6 @@
 			<Add option="-Wl,--enable-auto-import" />
 			<Add library="codeblocks" />
 			<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
-			<Add directory="bzip2" />
 			<Add directory="$(#WX30.lib)/gcc_dll$(WX_CFG)" />
 		</Linker>
 		<Unit filename="bzip2/blocksort.c">
diff --git a/src/plugins/contrib/devpak_plugin/DevPakPlugin_wx30_64.cbp b/src/plugins/contrib/devpak_plugin/DevPakPlugin_wx30_64.cbp
index 7f43620..a4a80d5 100644
--- a/src/plugins/contrib/devpak_plugin/DevPakPlugin_wx30_64.cbp
+++ b/src/plugins/contrib/devpak_plugin/DevPakPlugin_wx30_64.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="libbz2">
-				<Option output="bzip2/libbz2_64.a" prefix_auto="0" extension_auto="0" />
+				<Option output="../../../lib30_64/contrib/devpak_plugin/libbz2.a" prefix_auto="0" extension_auto="0" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/devpak_plugin" />
 				<Option type="2" />
@@ -46,7 +46,8 @@
 				</Compiler>
 				<Linker>
 					<Add option="-mthreads" />
-					<Add library="bz2_64" />
+					<Add library="bz2" />
+					<Add directory="../../../lib30_64/contrib/devpak_plugin" />
 					<Add directory="../../../devel30_64" />
 				</Linker>
 				<ExtraCommands>
@@ -75,7 +76,6 @@
 			<Add option="-Wl,--enable-auto-import" />
 			<Add library="codeblocks" />
 			<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
-			<Add directory="bzip2" />
 			<Add directory="$(#WX30_64.lib)/gcc_dll$(WX_CFG)" />
 		</Linker>
 		<Unit filename="bzip2/blocksort.c">
diff --git a/src/plugins/contrib/help_plugin/help-plugin.cbp b/src/plugins/contrib/help_plugin/help-plugin.cbp
index 7015570..1b7197a 100644
--- a/src/plugins/contrib/help_plugin/help-plugin.cbp
+++ b/src/plugins/contrib/help_plugin/help-plugin.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="bzip2">
-				<Option output="bzip2/libbzip2.a" prefix_auto="0" extension_auto="0" />
+				<Option output="../../../lib/contrib/help_plugin/libbzip2.a" prefix_auto="0" extension_auto="0" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs/plugins/contrib/help_plugin/bzip2" />
 				<Option type="2" />
@@ -22,7 +22,7 @@
 				</Linker>
 			</Target>
 			<Target title="zlib">
-				<Option output="zlib/libz.a" prefix_auto="0" extension_auto="0" />
+				<Option output="../../../lib/contrib/help_plugin/libz.a" prefix_auto="0" extension_auto="0" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs/plugins/contrib/help_plugin/zlib" />
 				<Option type="2" />
@@ -64,8 +64,9 @@
 				</Compiler>
 				<Linker>
 					<Add option="-mthreads" />
-					<Add library="bzip2\libbzip2.a" />
-					<Add library="zlib\libz.a" />
+					<Add library="libbzip2.a" />
+					<Add library="libz.a" />
+					<Add directory="../../../lib/contrib/help_plugin" />
 				</Linker>
 				<ExtraCommands>
 					<Add after="zip -j9 ..\..\..\devel\share\CodeBlocks\help_plugin.zip manifest.xml *.xrc *.png" />
diff --git a/src/plugins/contrib/help_plugin/help-plugin_wx30.cbp b/src/plugins/contrib/help_plugin/help-plugin_wx30.cbp
index 54a8521..669f4e7 100644
--- a/src/plugins/contrib/help_plugin/help-plugin_wx30.cbp
+++ b/src/plugins/contrib/help_plugin/help-plugin_wx30.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="bzip2">
-				<Option output="bzip2/libbzip2.a" prefix_auto="0" extension_auto="0" />
+				<Option output="../../../lib30/contrib/help_plugin/libbzip2.a" prefix_auto="0" extension_auto="0" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs30/plugins/contrib/help_plugin/bzip2" />
 				<Option type="2" />
@@ -22,7 +22,7 @@
 				</Linker>
 			</Target>
 			<Target title="zlib">
-				<Option output="zlib/libz.a" prefix_auto="0" extension_auto="0" />
+				<Option output="../../../lib30/contrib/help_plugin/libz.a" prefix_auto="0" extension_auto="0" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs30/plugins/contrib/help_plugin/zlib" />
 				<Option type="2" />
@@ -62,8 +62,9 @@
 				</Compiler>
 				<Linker>
 					<Add option="-mthreads" />
-					<Add library="bzip2\libbzip2.a" />
-					<Add library="zlib\libz.a" />
+					<Add library="libbzip2.a" />
+					<Add library="libz.a" />
+					<Add directory="../../../lib30/contrib/help_plugin" />
 				</Linker>
 				<ExtraCommands>
 					<Add after="zip -j9 ..\..\..\devel30\share\CodeBlocks\help_plugin.zip manifest.xml *.xrc *.png" />
diff --git a/src/plugins/contrib/help_plugin/help-plugin_wx30_64.cbp b/src/plugins/contrib/help_plugin/help-plugin_wx30_64.cbp
index a58b339..9ebb6df 100644
--- a/src/plugins/contrib/help_plugin/help-plugin_wx30_64.cbp
+++ b/src/plugins/contrib/help_plugin/help-plugin_wx30_64.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="bzip2">
-				<Option output="bzip2/libbzip2_64.a" prefix_auto="0" extension_auto="0" />
+				<Option output="../../../lib30_64/contrib/help_plugin/libbzip2.a" prefix_auto="0" extension_auto="0" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/help_plugin/bzip2" />
 				<Option type="2" />
@@ -22,7 +22,7 @@
 				</Linker>
 			</Target>
 			<Target title="zlib">
-				<Option output="zlib/libz_64.a" prefix_auto="0" extension_auto="0" />
+				<Option output="../../../lib30_64/contrib/help_plugin/libz.a" prefix_auto="0" extension_auto="0" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/help_plugin/zlib" />
 				<Option type="2" />
@@ -63,8 +63,9 @@
 				</Compiler>
 				<Linker>
 					<Add option="-mthreads" />
-					<Add library="bzip2\libbzip2_64.a" />
-					<Add library="zlib\libz_64.a" />
+					<Add library="libbzip2.a" />
+					<Add library="libz.a" />
+					<Add directory="../../../lib30_64/contrib/help_plugin" />
 				</Linker>
 				<ExtraCommands>
 					<Add after="zip -j9 ..\..\..\devel30_64\share\CodeBlocks\help_plugin.zip manifest.xml *.xrc *.png" />
diff --git a/src/plugins/contrib/regex_testbed/RegExTestbed.cbp b/src/plugins/contrib/regex_testbed/RegExTestbed.cbp
index f2213bd..cf869cb 100644
--- a/src/plugins/contrib/regex_testbed/RegExTestbed.cbp
+++ b/src/plugins/contrib/regex_testbed/RegExTestbed.cbp
@@ -10,7 +10,7 @@
 			<Target title="default">
 				<Option output="../../../devel/share/CodeBlocks/plugins/RegExTestbed" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/regex_testbed" />
-				<Option external_deps="../../../devel/libcodeblocks.a;" />
+				<Option external_deps="../../../lib/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --multiple-instance -na -ns -nd -p debug" />
diff --git a/src/plugins/contrib/regex_testbed/RegExTestbed_wx30.cbp b/src/plugins/contrib/regex_testbed/RegExTestbed_wx30.cbp
index a3470ac..8478bdc 100644
--- a/src/plugins/contrib/regex_testbed/RegExTestbed_wx30.cbp
+++ b/src/plugins/contrib/regex_testbed/RegExTestbed_wx30.cbp
@@ -10,7 +10,7 @@
 			<Target title="default">
 				<Option output="../../../devel30/share/CodeBlocks/plugins/RegExTestbed" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30/plugins/contrib/regex_testbed" />
-				<Option external_deps="../../../devel30/libcodeblocks.a;" />
+				<Option external_deps="../../../lib30/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --multiple-instance -na -ns -nd -p debug" />
diff --git a/src/plugins/contrib/regex_testbed/RegExTestbed_wx30_64.cbp b/src/plugins/contrib/regex_testbed/RegExTestbed_wx30_64.cbp
index 61f52a0..ff40cbb 100644
--- a/src/plugins/contrib/regex_testbed/RegExTestbed_wx30_64.cbp
+++ b/src/plugins/contrib/regex_testbed/RegExTestbed_wx30_64.cbp
@@ -10,7 +10,7 @@
 			<Target title="default">
 				<Option output="../../../devel30_64/share/CodeBlocks/plugins/RegExTestbed" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/regex_testbed" />
-				<Option external_deps="../../../devel30_64/libcodeblocks.a;" />
+				<Option external_deps="../../../lib30_64/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --multiple-instance -na -ns -nd -p debug" />
diff --git a/src/plugins/contrib/source_exporter/Exporter.cbp b/src/plugins/contrib/source_exporter/Exporter.cbp
index 7d2dfe9..39c7b20 100644
--- a/src/plugins/contrib/source_exporter/Exporter.cbp
+++ b/src/plugins/contrib/source_exporter/Exporter.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="wxPdfDocument">
-				<Option output="wxPdfDocument/lib/wxPdfDocument" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../lib/contrib/source_exporter/wxPdfDocument" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs/plugins/contrib/source_exporter" />
 				<Option type="2" />
@@ -58,7 +58,7 @@
 			<Target title="default">
 				<Option output="../../../devel/share/CodeBlocks/plugins/Exporter" prefix_auto="0" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/source_exporter" />
-				<Option external_deps="wxPdfDocument/lib/libwxPdfDocument.a;" />
+				<Option external_deps="../../../lib/contrib/source_exporter/libwxPdfDocument.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --multiple-instance -na -ns -nd -p debug" />
@@ -77,8 +77,8 @@
 					<Add library="wxPdfDocument" />
 					<Add library="wxmsw28$(WX_SUFFIX)" />
 					<Add library="gdi32" />
+					<Add directory="../../../lib/contrib/source_exporter" />
 					<Add directory="../../../devel" />
-					<Add directory="wxPdfDocument/lib" />
 				</Linker>
 				<ExtraCommands>
 					<Add after="zip -j9 ..\..\..\devel\share\CodeBlocks\Exporter.zip manifest.xml" />
diff --git a/src/plugins/contrib/source_exporter/Exporter_wx30.cbp b/src/plugins/contrib/source_exporter/Exporter_wx30.cbp
index 98abb1a..9f69674 100644
--- a/src/plugins/contrib/source_exporter/Exporter_wx30.cbp
+++ b/src/plugins/contrib/source_exporter/Exporter_wx30.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="wxPdfDocument">
-				<Option output="wxPdfDocument/lib30/wxPdfDocument" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../lib30/contrib/source_exporter/wxPdfDocument" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs30/plugins/contrib/source_exporter" />
 				<Option type="2" />
@@ -58,7 +58,7 @@
 			<Target title="default">
 				<Option output="../../../devel30/share/CodeBlocks/plugins/Exporter" prefix_auto="0" extension_auto="1" />
 				<Option object_output="../../../.objs30/plugins/contrib/source_exporter" />
-				<Option external_deps="wxPdfDocument/lib30/libwxPdfDocument.a;" />
+				<Option external_deps="../../../lib30/contrib/source_exporter/libwxPdfDocument.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --multiple-instance -na -ns -nd -p debug" />
@@ -77,8 +77,8 @@
 					<Add library="wxPdfDocument" />
 					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
 					<Add library="gdi32" />
+					<Add directory="../../../lib30/contrib/source_exporter" />
 					<Add directory="../../../devel30" />
-					<Add directory="wxPdfDocument/lib30" />
 				</Linker>
 				<ExtraCommands>
 					<Add after="zip -j9 ..\..\..\devel30\share\CodeBlocks\Exporter.zip manifest.xml" />
diff --git a/src/plugins/contrib/source_exporter/Exporter_wx30_64.cbp b/src/plugins/contrib/source_exporter/Exporter_wx30_64.cbp
index 961b603..5cfac51 100644
--- a/src/plugins/contrib/source_exporter/Exporter_wx30_64.cbp
+++ b/src/plugins/contrib/source_exporter/Exporter_wx30_64.cbp
@@ -8,7 +8,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="wxPdfDocument">
-				<Option output="wxPdfDocument/lib30_64/wxPdfDocument" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../lib30_64/contrib/source_exporter/wxPdfDocument" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/source_exporter" />
 				<Option type="2" />
@@ -58,7 +58,7 @@
 			<Target title="default">
 				<Option output="../../../devel30_64/share/CodeBlocks/plugins/Exporter" prefix_auto="0" extension_auto="1" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/source_exporter" />
-				<Option external_deps="wxPdfDocument/lib30_64/libwxPdfDocument.a;" />
+				<Option external_deps="../../../lib30_64/contrib/source_exporter/libwxPdfDocument.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --multiple-instance -na -ns -nd -p debug" />
@@ -78,7 +78,7 @@
 					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
 					<Add library="gdi32" />
 					<Add directory="../../../devel30_64" />
-					<Add directory="wxPdfDocument/lib30_64" />
+					<Add directory="../../../lib30_64/contrib/source_exporter" />
 				</Linker>
 				<ExtraCommands>
 					<Add after="zip -j9 ..\..\..\devel30_64\share\CodeBlocks\Exporter.zip manifest.xml" />
diff --git a/src/plugins/contrib/wxContribItems/wxContribItems.cbp b/src/plugins/contrib/wxContribItems/wxContribItems.cbp
index 8d732db..0991763 100644
--- a/src/plugins/contrib/wxContribItems/wxContribItems.cbp
+++ b/src/plugins/contrib/wxContribItems/wxContribItems.cbp
@@ -9,7 +9,7 @@
 		<Build>
 			<Target title="wxFlatNoteBook">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxflatnotebook" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxflatnotebook" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/wxFlatNotebook" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -21,7 +21,7 @@
 			</Target>
 			<Target title="wxChartCtrl">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxchartctrl" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxchartctrl" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/wxChartCtrl" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -33,7 +33,7 @@
 			</Target>
 			<Target title="wxCustomButton">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxcustombutton" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxcustombutton" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/wxthings" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -45,7 +45,7 @@
 			</Target>
 			<Target title="wxImagePanel">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wximagepanel" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wximagepanel" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/wxImagePanel" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -59,7 +59,7 @@
 			</Target>
 			<Target title="wxSpeedButton">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxspeedbutton" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxspeedbutton" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/wxSpeedButton" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -70,7 +70,7 @@
 			</Target>
 			<Target title="wxKWIC">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxkwic" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxkwic" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/KWIC" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -84,7 +84,7 @@
 			</Target>
 			<Target title="wxTreeList">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxtreelist" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxtreelist" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/wxtreelist" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -95,7 +95,7 @@
 			</Target>
 			<Target title="wxLED">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxled" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxled" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs/plugins/contrib/wxContribItems/wxled" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -131,7 +131,6 @@
 			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#wx.include)" />
 			<Add directory="$(#wx.lib)/gcc_dll$(WX_CFG)/msw$(WX_SUFFIX)" />
-			<Add directory="wxFlatNotebook/include" />
 		</Compiler>
 		<Linker>
 			<Add option="-Wl,--enable-auto-import" />
diff --git a/src/plugins/contrib/wxContribItems/wxContribItems_wx30.cbp b/src/plugins/contrib/wxContribItems/wxContribItems_wx30.cbp
index 8dc7384..84b0def 100644
--- a/src/plugins/contrib/wxContribItems/wxContribItems_wx30.cbp
+++ b/src/plugins/contrib/wxContribItems/wxContribItems_wx30.cbp
@@ -9,7 +9,7 @@
 		<Build>
 			<Target title="wxFlatNoteBook">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30/wxflatnotebook" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30/wxflatnotebook" imp_lib="../../../lib30/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30/plugins/contrib/wxContribItems/wxFlatNotebook" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -21,7 +21,7 @@
 			</Target>
 			<Target title="wxChartCtrl">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30/wxchartctrl" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30/wxchartctrl" imp_lib="../../../lib30/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30/plugins/contrib/wxContribItems/wxChartCtrl" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -33,7 +33,7 @@
 			</Target>
 			<Target title="wxCustomButton">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30/wxcustombutton" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30/wxcustombutton" imp_lib="../../../lib30/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30/plugins/contrib/wxContribItems/wxthings" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -45,7 +45,7 @@
 			</Target>
 			<Target title="wxImagePanel">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30/wximagepanel" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30/wximagepanel" imp_lib="../../../lib30/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30/plugins/contrib/wxContribItems/wxImagePanel" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -59,7 +59,7 @@
 			</Target>
 			<Target title="wxSpeedButton">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30/wxspeedbutton" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30/wxspeedbutton" imp_lib="../../../lib30/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30/plugins/contrib/wxContribItems/wxSpeedButton" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -70,7 +70,7 @@
 			</Target>
 			<Target title="wxKWIC">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30/wxkwic" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30/wxkwic" imp_lib="../../../lib30/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30/plugins/contrib/wxContribItems/KWIC" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -84,7 +84,7 @@
 			</Target>
 			<Target title="wxLED">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30/wxled" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30/wxled" imp_lib="../../../lib30/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30/plugins/contrib/wxContribItems/wxled" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -120,13 +120,11 @@
 			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX30.include)" />
 			<Add directory="$(#WX30.lib)/gcc_dll$(WX_CFG)/msw$(WX_SUFFIX)" />
-			<Add directory="wxFlatNotebook/include" />
 		</Compiler>
 		<Linker>
 			<Add option="-Wl,--enable-auto-import" />
 			<Add option="-mthreads" />
 			<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
-			<Add directory="../../../devel30" />
 			<Add directory="$(#WX30.lib)/gcc_dll$(WX_CFG)" />
 		</Linker>
 		<Unit filename="KWIC/include/wx/KWIC/AngularMeter.h">
diff --git a/src/plugins/contrib/wxContribItems/wxContribItems_wx30_64.cbp b/src/plugins/contrib/wxContribItems/wxContribItems_wx30_64.cbp
index 9cdd776..1147df0 100644
--- a/src/plugins/contrib/wxContribItems/wxContribItems_wx30_64.cbp
+++ b/src/plugins/contrib/wxContribItems/wxContribItems_wx30_64.cbp
@@ -9,7 +9,7 @@
 		<Build>
 			<Target title="wxFlatNoteBook">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30_64/wxflatnotebook" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30_64/wxflatnotebook" imp_lib="../../../lib30_64/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/wxContribItems/wxFlatNotebook" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -21,7 +21,7 @@
 			</Target>
 			<Target title="wxChartCtrl">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30_64/wxchartctrl" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30_64/wxchartctrl" imp_lib="../../../lib30_64/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/wxContribItems/wxChartCtrl" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -33,7 +33,7 @@
 			</Target>
 			<Target title="wxCustomButton">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30_64/wxcustombutton" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30_64/wxcustombutton" imp_lib="../../../lib30_64/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/wxContribItems/wxthings" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -45,7 +45,7 @@
 			</Target>
 			<Target title="wxImagePanel">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30_64/wximagepanel" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30_64/wximagepanel" imp_lib="../../../lib30_64/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/wxContribItems/wxImagePanel" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -59,7 +59,7 @@
 			</Target>
 			<Target title="wxSpeedButton">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30_64/wxspeedbutton" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30_64/wxspeedbutton" imp_lib="../../../lib30_64/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/wxContribItems/wxSpeedButton" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -70,7 +70,7 @@
 			</Target>
 			<Target title="wxKWIC">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30_64/wxkwic" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30_64/wxkwic" imp_lib="../../../lib30_64/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/wxContribItems/KWIC" />
 				<Option type="3" />
 				<Option compiler="gcc" />
@@ -84,7 +84,7 @@
 			</Target>
 			<Target title="wxLED">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30_64/wxled" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30_64/wxled" imp_lib="../../../lib30_64/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/wxContribItems/wxled" />
 				<Option type="3" />
 				<Option compiler="gcc" />
diff --git a/src/plugins/contrib/wxSmith/wxSmith.cbp b/src/plugins/contrib/wxSmith/wxSmith.cbp
index f0e656c..905895e 100644
--- a/src/plugins/contrib/wxSmith/wxSmith.cbp
+++ b/src/plugins/contrib/wxSmith/wxSmith.cbp
@@ -9,7 +9,7 @@
 		<Build>
 			<Target title="wxSmithLib">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel/wxsmithlib" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel/wxsmithlib" imp_lib="../../../lib/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="../../../devel" />
 				<Option object_output="../../../.objs/plugins/contrib/wxSmithLib" />
 				<Option type="3" />
diff --git a/src/plugins/contrib/wxSmith/wxSmith_wx30.cbp b/src/plugins/contrib/wxSmith/wxSmith_wx30.cbp
index 4a70ce6..4134e10 100644
--- a/src/plugins/contrib/wxSmith/wxSmith_wx30.cbp
+++ b/src/plugins/contrib/wxSmith/wxSmith_wx30.cbp
@@ -9,7 +9,7 @@
 		<Build>
 			<Target title="wxSmithLib">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30/wxsmithlib" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30/wxsmithlib" imp_lib="../../../lib30/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="../../../devel30" />
 				<Option object_output="../../../.objs30/plugins/contrib/wxSmithLib" />
 				<Option type="3" />
diff --git a/src/plugins/contrib/wxSmith/wxSmith_wx30_64.cbp b/src/plugins/contrib/wxSmith/wxSmith_wx30_64.cbp
index 4feaf66..6a45106 100644
--- a/src/plugins/contrib/wxSmith/wxSmith_wx30_64.cbp
+++ b/src/plugins/contrib/wxSmith/wxSmith_wx30_64.cbp
@@ -9,7 +9,7 @@
 		<Build>
 			<Target title="wxSmithLib">
 				<Option platforms="Windows;" />
-				<Option output="../../../devel30_64/wxsmithlib" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../devel30_64/wxsmithlib" imp_lib="../../../lib30_64/$(TARGET_OUTPUT_BASENAME)" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME)" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="../../../devel30_64" />
 				<Option object_output="../../../.objs30_64/plugins/contrib/wxSmithLib" />
 				<Option type="3" />
diff --git a/src/plugins/contrib/wxSmithPlot/wxsSmithPlot.cbp b/src/plugins/contrib/wxSmithPlot/wxsSmithPlot.cbp
index 89be967..1d1d83a 100644
--- a/src/plugins/contrib/wxSmithPlot/wxsSmithPlot.cbp
+++ b/src/plugins/contrib/wxSmithPlot/wxsSmithPlot.cbp
@@ -7,7 +7,7 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="MathPlot">
-				<Option output="../../../devel/MathPlot" prefix_auto="1" extension_auto="1" />
+				<Option output="../../../lib/contrib/wxSmithPlot/MathPlot" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
 				<Option object_output="../../../.objs/plugins/contrib/wxSmithPlot/MathPlot" />
 				<Option type="2" />
@@ -42,6 +42,7 @@
 					<Add library="wxsmithlib" />
 					<Add library="codeblocks" />
 					<Add library="wxpropgrid" />
+					<Add directory="../../../lib/contrib/wxSmithPlot" />
 					<Add directory="../../../devel" />
 				</Linker>
 				<ExtraCommands>
diff --git a/src/tools/cb_share_config/cb_share_config.cbp b/src/tools/cb_share_config/cb_share_config.cbp
index 44881f1..5dada02 100644
--- a/src/tools/cb_share_config/cb_share_config.cbp
+++ b/src/tools/cb_share_config/cb_share_config.cbp
@@ -46,7 +46,7 @@
 			<Add library="txml" />
 			<Add library="wxmsw28$(WX_SUFFIX)" />
 			<Add directory="$(#WX.lib)/gcc_dll$(WX_CFG)" />
-			<Add directory="../../base/tinyxml" />
+			<Add directory="../../lib" />
 		</Linker>
 		<Unit filename="app.cpp" />
 		<Unit filename="app.h" />
diff --git a/src/tools/cb_share_config/cb_share_config_wx30.cbp b/src/tools/cb_share_config/cb_share_config_wx30.cbp
index 23a0b97..14ef096 100644
--- a/src/tools/cb_share_config/cb_share_config_wx30.cbp
+++ b/src/tools/cb_share_config/cb_share_config_wx30.cbp
@@ -46,7 +46,7 @@
 			<Add library="txml" />
 			<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
 			<Add directory="$(#WX30.lib)/gcc_dll$(WX_CFG)" />
-			<Add directory="../../base/tinyxml" />
+			<Add directory="../../lib30" />
 		</Linker>
 		<Unit filename="app.cpp" />
 		<Unit filename="app.h" />
diff --git a/src/tools/cb_share_config/cb_share_config_wx30_64.cbp b/src/tools/cb_share_config/cb_share_config_wx30_64.cbp
index b093e3e..9c4d1b5 100644
--- a/src/tools/cb_share_config/cb_share_config_wx30_64.cbp
+++ b/src/tools/cb_share_config/cb_share_config_wx30_64.cbp
@@ -46,10 +46,10 @@
 		</ResourceCompiler>
 		<Linker>
 			<Add option="-mthreads" />
-			<Add library="txml30_64" />
+			<Add library="txml" />
 			<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
 			<Add directory="$(#WX30_64.lib)/gcc_dll$(WX_CFG)" />
-			<Add directory="../../base/tinyxml" />
+			<Add directory="../../lib30_64" />
 		</Linker>
 		<Unit filename="app.cpp" />
 		<Unit filename="app.h" />
-- 
2.5.1.windows.1

